src/ua.c

```diff
@@ -1239,6 +1239,60 @@ int ua_debug(struct re_printf *pf, const struct ua *ua)
 }
 
 
+/**
+ * Print the user-agent information in JSON
+ *
+ * @param od  User-Agent dict
+ * @param pf  Print function
+ * @param ua  User-Agent object
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int ua_state_json_api(struct odict *od, struct re_printf *pf, const struct ua *ua)
+{
+	struct odict *reg = NULL;
+	struct odict *cfg = NULL;
+	struct le *le;
+	int err = 0;
+
+	if (!ua)
+		return 0;
+
+	err |= odict_alloc(&reg, 8);
+	err |= odict_alloc(&cfg, 8);
+
+	/* user-agent info */
+	err |= odict_entry_add(od, "cuser", ODICT_STRING, ua->cuser);
+
+	/* account info */
+	err |= account_json_api(od, cfg, ua->acc);
+	if (err)
+		warning("ua: failed to encode json account (%m)\n", err);
+
+	/* registration info */
+	err |= odict_entry_add(reg, "interval", ODICT_INT, ua->acc->regint);
+	err |= odict_entry_add(reg, "q_value", ODICT_DOUBLE, ua->acc->regq);
+	/* TODO: current not working */
+	err |= odict_entry_add(reg, "current", ODICT_BOOL, ua == uag_current());
+
+	for (le = ua->regl.head; le; le = le->next) {
+		/* TODO: how to get only current ua register state? */
+		err |= reg_json_api(reg, le->data);
+	}
+	if (err)
+		warning("ua: failed to encode json registration (%m)\n", err);
+
+	/* package */
+	/* TODO: this is complaining, how to build these objects? */
```

Comment:
The adding of:
* odict_entry_add(od, "settings", ODICT_OBJECT, cfg)
* odict_entry_add(od, "registration", ODICT_OBJECT, reg)
is causing Invalid argument, but seems to do the job, I'm missing something